weave/init_message.py

```diff
@@ -56,12 +56,28 @@ def _print_version_check() -> None:
             print(use_message)
 
 
+def _assert_min_version(host: str, min_required_version: str) -> None:
+    import weave
+
+    if _parse_version(min_required_version) > _parse_version(weave.__version__):
+        message = (
+            f"The target Weave host {host} requires a `weave` package version >= {min_required_version}."
```

Comment:
You can also not call weave.init() and your code will run normally without logging. Maybe say that in the PR in case they don't want to upgrade right away?

Reply:
added this in an upgrade message